### ISSUE TITLE
Issues with time in queries when creating alarms from SPY

### ISSUE THREAD

**Issue by havlan** [sirensolutions/sentinl#454]
1. Sentinl and Kibana version 
Kibana: 6.2.4
Sentinl: 6.2.4 (tag 6.2.3.3)

2. Issue description
Creating alarms with SPY on visualizations ends with passing the specific timestamps and not the dynamic time fields. This removes the dynamic aspect of a query.

3. Reproducing the issue step-by-step
Creating a line visualization.
Time field, upper right corner on kibana:
![screenshot from 2018-06-13 09-24-24](https://user-images.githubusercontent.com/18334546/41336157-aba9e652-6eeb-11e8-9aec-4ab95db23ec8.png)
- Index: mos-*
- Y-axis: count
- X-axis: Date Range
   - From: now-1w/w
   - To: now/w

4. Expected results vs. actual results
[Actual.txt](https://gist.githubusercontent.com/havlan/ac06509acdae14c0ac1bd06a8751722a/raw/011dee85362a5b96be36ea8be834ce84555c212c/time_actual.txt)
[Expected.txt](https://gist.githubusercontent.com/havlan/ac06509acdae14c0ac1bd06a8751722a/raw/011dee85362a5b96be36ea8be834ce84555c212c/time_expected.txt)

TLDR: 
Actual: `{"gte":1528869894415,"lte":1528873494415,"format":"epoch_millis"}`
Expected: `{"gte":"now-1h/h","lte":"now/h"}`

5. Sample of data
Script to generate data: [mos.sh](https://gist.githubusercontent.com/havlan/ac06509acdae14c0ac1bd06a8751722a/raw/dafa1e66a90273799228cbe3307ee039feb7f729/mos.sh)
6. Additional details

**Comment by sergibondarenko** [comment]
The time is the same, it is in milliseconds since the epoch.
```
1528873494415 - 1528869894415 = 3600000 ms (1 hour)
```

**Comment by havlan** [comment]
I am aware of that, but this makes the query generated from the visualization only valid for one specific hour.  If the timestamps are passed on to a watcher the queries will not be dynamic. The watcher relies upon someone manually making the queries dynamic.

**Comment by lmangani** [comment]
@havlan I agree with this proposal, it'll make the wizard smarter.

Does the elasticsearch API allow for dates formats such as ```now-180m/m```? 
If so we can use minutes as base measure and use something as follows?

###### mock
```
function diff_minutes(dt2, dt1) 
 {
  var diff =(dt2.getTime() - dt1.getTime()) / 1000;
  diff /= 60;
  return Math.abs(Math.round(diff));
 }

var input = {"gte":1528869894415,"lte":1528873494415,"format":"epoch_millis"}
var diff = diff_minutes( new Date(inpute.gte), new Date(inpute.lte)) + 'm';
var timerange = {"gte":"now-"+diff,"lte":"now/h"}
```
```
{
  gte: "now-60m/m",
  lte: "now/m"
}
```

**Comment by havlan** [comment]
@lmangani  I think so (manually tested). I got a suggestion using seconds coming up in a sec.

**Comment by havlan** [comment]
@lmangani #457

**Comment by havlan** [comment]
https://github.com/havlan/sentinl/tree/454_dynamic_time
It's a simple solution, it solves the "general" case of `@timestamp` usage. But I don't know if it breaks anything else. Are there more time fields that should be included? Or is this an ok default?

**Comment by sergibondarenko** [comment]
@havlan do you mean you want to add a possibility to specify a custom time field?

**Comment by havlan** [comment]
@sergibondarenko I was hoping to add a solution that does not assume default '@timestamp'.
```javascript
if (indexPattern && indexPattern.getTimeField()) {
        if (!alarm._source.input.search.request.body.query.bool.must) {
          alarm._source.input.search.request.body.query.bool.must = [];
        }

        alarm._source.input.search.request.body.query.bool.must.push({
          range: {
            [indexPattern.timeFieldName]: {
              from: watcherRange
            }
          }
        });
      }
```
I think if this if would ever pass, I don't need to specify which time field to replace, some code is still required though. Would make code cleaner and easier to eventually add more timefields.
```javascript 
if(must[key].range['@timestamp'])
```
```javascript
switch(indexPattern.timeFieldName)
```
I would still need to manually create edits for each common timestamp field, I guess?

**Comment by lmangani** [comment]
@havlan an override option would always be great but i 100% agree with your approach here

**Comment by lmangani** [comment]
@havlan feel free to override anything specific in your PR towards the goal, we'll review and test the overall and potential side-effects together afterwards, how's that?

**Comment by havlan** [comment]
@lmangani my thought is that in order to achieve this I need the ```indexPattern``` to somehow be set or passed along. It is not being set right now.
